partner_multi_company/models/res_users.py

```diff
@@ -12,16 +12,28 @@ def create(self, vals):
         res = super(ResUsers, self).create(vals)
         if "company_ids" in vals:
             res.partner_id.company_ids = vals["company_ids"]
-        if "company_id" in vals and res.partner_id.company_ids:
```

Comment:
Why removing this?

Reply:
Because this was totally legacy, form the time in which the activa company was self.env.user.company_id, since v13 this doesn't make sense anymore as self.env.user.company_id is the default company that you should not touch much, and the environment company is what matters (self.env.company)

Ref https://github.com/OCA/multi-company/issues/182

Reply:
Well, not really, as this is only on the `create` method, so switching between companies doesn't affect this, but well, with the other reasoning of keeping empty the companies on the partner (if we go there), this should be removed indeed.

Reply:
Yes, I see what you say and I agree that partner companies should be kept empty. Actually the meaning of company_id in user is not exactly the same as in other models since v13.